src/validators/dict.rs

```diff
@@ -147,4 +150,78 @@ macro_rules! build_validate {
 impl DictValidator {
     build_validate!(validate_dict, PyDict);
     build_validate!(validate_json_object, JsonObject);
+
+    fn validate_mapping<'s, 'data>(
```

Comment:
Another copy-pasted code block for the same reason above.

Reply:
looks like it would be worth either a macro or you might be able to use an `impl` to avoid duplication.

Reply:
Let me try to use `impl` first.